apps/meteor/ee/server/lib/deviceManagement/session.ts

```diff
@@ -23,88 +22,102 @@ Meteor.startup(() => {
 const uaParser = async (
 	uaString: ISocketConnection['httpHeaders']['user-agent'],
 ): Promise<UAParser.IResult & { app?: { name: string; version: string; bundle: string } }> => {
-	let rcAgent = {};
-	if (uaString && UAParserMobile.isMobileApp(uaString)) {
-		rcAgent = UAParserMobile.uaObject(uaString);
-	}
-
-	if (uaString && UAParserDesktop.isDesktopApp(uaString)) {
-		rcAgent = UAParserDesktop.uaObject(uaString);
-	}
-
 	const ua = new UAParser(uaString);
-	return { ...ua.getResult(), ...rcAgent };
+	return {
+		...ua.getResult(),
+		...(uaString && UAParserMobile.isMobileApp(uaString) && UAParserMobile.uaObject(uaString)),
+		...(uaString && UAParserDesktop.isDesktopApp(uaString) && UAParserDesktop.uaObject(uaString)),
+	};
 };
 
-export const listenSessionLogin = async (): Promise<void> => {
-	deviceManagementEvents.on('device-login', async ({ userId, connection }) => {
-		if (!hasLicense('device-management')) return;
+export const listenSessionLogin = () => {
+	return deviceManagementEvents.on('device-login', async ({ userId, connection }) => {
+		const deviceEnabled = settings.get('Device_Management_Enable_Login_Emails');
+
+		if (!deviceEnabled) {
+			return;
+		}
 
-		const user = await Users.findOneById<IUser>(userId, { projection: { name: 1, username: 1, emails: 1 } });
-		if (user?.emails?.length && !connection.loginToken) {
-			const {
-				name,
-				username,
-				emails: [{ address: email }],
-			} = user;
-			const { browser, os, device, cpu, app } = await uaParser(connection.httpHeaders['user-agent']);
+		const user = await Users.findOne(
```

Comment:
can you move this one to the model?

Reply:
I can, do you have any suggestion? findOneByIdAndContainsEmailAndThatPreference :p